pgmpy/factors/Factor.py

```diff
@@ -4,6 +4,7 @@
 from collections import OrderedDict
 
 import numpy as np
+from sklearn.utils.extmath import cartesian
```

Comment:
Please don't add any other dependencies unless we can't deal with it. ```cartesian``` method is not a very complicated method and can be implemented very easily in ```pgmpy``` as well. So adding ```sklearn``` as a dependency just for this method doesn't make any sense.